libr/core/task.c

```diff
@@ -193,6 +197,9 @@ R_API void r_core_task_schedule(RCoreTask *current, RTaskState next_state) {
 
 static void task_wakeup(RCoreTask *current) {
 	RCore *core = current->core;
+	if (r_list_length (core->tasks) == 1) {
+		return;
```

Comment:
This will break things. For example: The main task starts, core->tasks has length 1, so its state is never changed to R_CORE_TASK_STATE_RUNNING. Another tasks starts while the main task is still running and it will start executing without waiting to be scheduled, because it does not detect the main task is still running.